proxywasm/vmstate.go

```diff
@@ -57,6 +59,14 @@ func SetNewStreamContext(f func(rootContextID, contextID uint32) StreamContext)
 	currentState.newStreamContext = f
 }
 
+func GetRootContextByID(rootContextID uint32) (RootContext, error) {
+	rootContextState, ok := currentState.rootContexts[rootContextID]
+	if !ok {
+		return nil, types.ErrorStatusNotFound
```

Comment:
types.ErrorStatusNotFound corresponds to the error status code returned by the "host" in any hostcall
- https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/proxywasm/types/types.go#L67
 - https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/context.cc#L74
 
so I think we can just do like: 
```suggestion
		return nil, fmt.Errorf("root context not found: %d", rootContextID)
```

Reply:
Maybe it would be better to define a static error then:
`var ErrorRootContextNotFound = errors.New("root context not found")`
and then wrap it with `fmt.Errorf("%w: %d", rootContextID)`? That way, the error can be identified by the caller using `errors.Is`.

If you agree: I'm not sure where you would want to place this static error, though; there does not seem to be a standard for it in this project yet. Do you have a preference, or should I just place it above the function?

Reply:
agree with static error! Go ahead :+1:

Reply:
Alright, I've added it. I've also added a unit test, unfortunately I can't seem to get the e2e tests to run locally.